sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -255,7 +255,8 @@ namespace Azure { namespace Storage { namespace Blobs {
     }
     firstChunkLength = std::min(firstChunkLength, blobRangeSize);
 
-    if (static_cast<size_t>(blobRangeSize) > bufferSize)
+    if (static_cast<uint64_t>(blobRangeSize) > std::numeric_limits<size_t>::max()
```

Comment:
So, I am guessing we are casting `blobRangeSize` to uint64_t to also validate that the int64_t wasn't negative. Is that needed? Is it possible for it to be < 0, and if so, consider adding a comment here to explain that. since it isn't obvious (though I do like this way of validating two conditions at the same time).

Reply:
> I am guessing we are casting blobRangeSize to uint64_t to also validate that the int64_t wasn't negative. Is that needed?

No, that wasn't my intention. This is to check casting `blobRangeSize` to `size_t` won't lead to overflow.